core/shared/src/main/scala/zio/ZManaged.scala

```diff
@@ -926,6 +945,10 @@ object ZManaged {
       ZManaged.environment.flatMap(f)
   }
 
+  trait Preallocate {
```

Comment:
How about:
```scala
trait Scope {
  def allocate
}
```
?

Reply:
I think naming this scope is better, Preallocate name is a remnant from previous naming.
I don't know about allocate not being apply. I think this would often become more clumsy:
```scala
val resource: Managed[Nothing, Int] = ???

ZManaged.scope.use { scope =>
  scope.preallocate(resource)
} 
```

Reply:
Ok, agreed about `apply` - sounds good to me.